### PR TITLE
tests,vm-virtio,vmm: Use 'socket' for all CLI/API parameters

### PR DESCRIPTION
**Pull request by likebreath**
This patch unifies the inconsistent uses of 'socket' and 'sock' from our
CLI/API parameters.

Fixes: #1091

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Comment by rbradford** [comment]
This is great. For the vhost_user_fs daemon binary please make it accept both the old and new names giving a deprecation warning for the old.

This is because I think other folks may be using that binary.

We can complete the rename after the release by removing then.

File: src/bin/vhost_user_fs.rs [hunk 1]
```diff
@@ -295,8 +295,8 @@ fn main() {
                 .min_values(1),
         )
         .arg(
-            Arg::with_name("sock")
-                .long("sock")
+            Arg::with_name("socket")
+                .long("socket")
```
Comment (rbradford) [inline src/bin/vhost_user_fs.rs:299]: This is the only one where I think we should temporarily support both parameters.

Comment (likebreath) [reply src/bin/vhost_user_fs.rs:306]: Thanks for the comments and pointers. I just updated the PR, and that seems to serve the purpose. Please let me know if there is better ways to do it. :)

File: vmm/src/config.rs [hunk 2]
```diff
@@ -1749,27 +1749,27 @@ mod tests {
         );
         // DAX on -> default cache size
         assert_eq!(
-            FsConfig::parse("tag=mytag,sock=/tmp/sock,dax=on")?,
+            FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=on")?,
             FsConfig {
-                sock: PathBuf::from("/tmp/sock"),
+                socket: PathBuf::from("/tmp/socket"),
```
Comment (sboeuf) [inline vmm/src/config.rs:1754]: Ditto

File: vmm/src/config.rs [hunk 3]
```diff
@@ -1749,27 +1749,27 @@ mod tests {
...
                 tag: "mytag".to_owned(),
                 dax: true,
                 cache_size: default_fsconfig_cache_size(),
                 ..Default::default()
             }
         );
         assert_eq!(
-            FsConfig::parse("tag=mytag,sock=/tmp/sock,dax=on,cache_size=4G")?,
+            FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=on,cache_size=4G")?,
```
Comment (sboeuf) [inline vmm/src/config.rs:1762]: Ditto

File: vmm/src/config.rs [hunk 4]
```diff
@@ -1749,27 +1749,27 @@ mod tests {
...
             FsConfig {
-                sock: PathBuf::from("/tmp/sock"),
+                socket: PathBuf::from("/tmp/socket"),
```
Comment (sboeuf) [inline vmm/src/config.rs:1764]: Ditto

File: vmm/src/config.rs [hunk 5]
```diff
@@ -1749,27 +1749,27 @@ mod tests {
...
                 tag: "mytag".to_owned(),
                 dax: true,
                 cache_size: 4 << 30,
                 ..Default::default()
             }
         );
         // Cache size without DAX is an error
-        assert!(FsConfig::parse("tag=mytag,sock=/tmp/sock,dax=off,cache_size=4G").is_err());
+        assert!(FsConfig::parse("tag=mytag,socket=/tmp/socket,dax=off,cache_size=4G").is_err());
```
Comment (sboeuf) [inline vmm/src/config.rs:1772]: Ditto

**Comment by likebreath** [comment]
@sboeuf I have updated the changes based on your comments. I also cleaned up few places from `main.rs` and `config.rs` so that we only use "sock" (and not "socket") from file paths. For example: changes from `src/main.rs:748`. Any more advices, please let me know. Thanks!

**Comment by sboeuf** [comment]
Thanks @likebreath

**Comment by rbradford** [comment]
@sboeuf did @likebreath address all your comments?

**Pull request merged**